### ISSUE TITLE
cross-validated rmsd k-medoids without hogging memory?

### ISSUE THREAD

**Issue by kyleabeauchamp** [msmbuilder/osprey#82]
I was hoping to use MiniBatchKMedoids with the RMSD metric in Osprey.  Am I correct in believing that the 5fold cross-validation requires loading all of the data into memory up front?  For example, the following (non-cross validated) script seems to use lazy loading and low memory:

```
from msmbuilder import example_datasets, cluster, msm, featurizer, lumping, utils, dataset, decomposition
from sklearn.pipeline import make_pipeline

n_clusters = 2

trajectories = dataset.MDTrajDataset("./trajectories/*.h5")

clusterer = cluster.MiniBatchKMedoids(n_clusters=n_clusters, metric="rmsd")
clusterer.fit(trajectories)
```

**Comment by rmcgibbo** [comment]
I don't think it _should_ be necessary to load everything into memory simultaneously if you're using the msmbuilder dataset loader in osprey. It could be accidental. Can you figure out where it's happening?

**Comment by kyleabeauchamp** [comment]
Here's the config file:

```
estimator:
    pickle: my-model.pkl
    entry_point: msmbuilder.decomposition.tICA
    eval: |
        Pipeline([
                ("clusterer", MiniBatchKMedoids(batch_size=5, metric="rmsd")),
                ("msm", MarkovStateModel(n_timescales=8, lag_time=400))                
        ])
    eval_scope: msmbuilder


strategy:
    name: hyperopt_tpe  # or moe, hyperopt_tpe

search_space:
  clusterer__n_clusters:
    min: 10
    max: 50
    type: int

cv: 5  # the order of K-fold cross validation to use

dataset_loader:
  name: mdtraj
  params:
    trajectories: ../trajectories/*.h5
    

trials:
  uri: sqlite:///kmeans.db
```

**Comment by rmcgibbo** [comment]
you need to switch your dataset loader from `mdtraj` to `msmbuilder`. currently, you're going through this code path: https://github.com/pandegroup/osprey/blob/master/osprey/dataset_loaders.py#L58-L62

**Comment by rmcgibbo** [comment]
but you want to be up here: https://github.com/pandegroup/osprey/blob/master/osprey/dataset_loaders.py#L16-L29

**Comment by kyleabeauchamp** [comment]
OK, I updated the yaml file according to your suggestion (see edited code block above), but I still see huge memory usage (see image):

![memory](https://cloud.githubusercontent.com/assets/1900459/7600113/586f2de6-f8d6-11e4-930e-b16bcf68d1e0.png)

**Comment by kyleabeauchamp** [comment]
The drop at the end is when I control-C the osprey run and clear up my memory.
